transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -611,8 +611,8 @@ private static void checkMultiplicity(ChannelHandler handler) {
 
     private void callHandlerAdded0(final AbstractChannelHandlerContext ctx) {
         try {
-            ctx.handler().handlerAdded(ctx);
             ctx.setAddComplete();
+            ctx.handler().handlerAdded(ctx);
```

Comment:
just do me a favour and add a comment that explains the ordering. Otherwise we may forget 👍

Reply:
done